jobs/build/build-sync/Jenkinsfile

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env groovy
+
+node {
+    checkout scm
+    def buildlib = load("pipeline-scripts/buildlib.groovy")
+    def commonlib = buildlib.commonlib
+
+    properties(
+        [
+            buildDiscarder(
+                logRotator(
+                    artifactDaysToKeepStr: '',
+                    artifactNumToKeepStr: '',
+                    daysToKeepStr: '',
+                    numToKeepStr: '360'
+                )
+            ),
+            [
+                $class : 'ParametersDefinitionProperty',
+                parameterDefinitions: [
+                    commonlib.mockParam(),
+		    commonlib.ocpVersionParam('BUILD_VERSION', '4'),
+                ]
+            ],
+            disableConcurrentBuilds(),
+            timestamps(),
+        ]
+    )
+
+    echo "Initializing ${params.BUILD_VERSION} sync: #${currentBuild.number}"
+
+    // doozer_working must be in WORKSPACE in order to have artifacts archived
+    def mirrorWorking = "${WORKSPACE}/MIRROR_working"
+    // Location of the SRC=DEST input file
+    def ocMirrorInput = "${mirrorWorking}/oc_mirror_input"
+    // Location of the image stream to apply
+    def ocIsObject = "${mirrorWorking}/release-is.yaml"
+    // Locally stored image stream stub
+    def baseImageStream = "/home/jenkins/base-art-latest-imagestream.yaml"
+    // Kubeconfig allowing ART to interact with api.ci.openshift.org
+    def ciKubeconfig = "/home/jenkins/kubeconfigs/art-publish.kubeconfig"
+
+    // See 'oc image mirror --help' for more information.
+    // This is the template for the SRC=DEST strings mentioned above.
+    def ocFmtStr = "registry.reg-aws.openshift.com:443/{repository}=quay.io/openshift-release-dev/ocp-v4.0-art-dev:{version}-{release}-{image_name_short}"
+
+    //Clear out previous work
+    sh "rm -rf ${mirrorWorking}"
+    sh "mkdir -p ${mirrorWorking}"
+
+    stage("Version dumps") {
+        buildlib.doozer """
+--version
+"""
+        sh "which doozer"
+        sh "oc version"
+    }
+
+    // TRY all of this so we can save the generated artifacts before
+    // throwing the exceptions
+    try {
+        // ######################################################################
+        // This should create a list of SOURCE=DEST strings in the output file
+        // May take a few minutes because doozer must query brew for each image
+        stage("Generate SRC=DEST input") {
+            buildlib.doozer """
+--working-dir "${mirrorWorking}" --group 'openshift-${params.BUILD_VERSION}'
+beta:release-gen
+--sd ${ocMirrorInput}
+--is ${ocIsObject}
+--isb ${baseImageStream}
```

Comment:
do these opts have long-form? would help with knowing what this does

Reply:
Will do!